python/lsst/analysis/tools/atools/diaSourceMetrics.py

```diff
@@ -76,3 +80,74 @@ def setDefaults(self):
 
         # the units for the quantity (count, an astropy quantity)
         self.produce.metric.units = {"numDipoles": "ct"}
+
+
+class NumDiaSourcesHighReliabilityMetric(AnalysisTool):
```

Comment:
This, and the next two, AnalysisTools are not done in a wrong way per-say, but they kind of break the spirit of what AnalysisTools allow you to do.

Because these tools only differ in configuration (if you accept my recommendations above), then it is possible to have a single tool, perhapse called something like `NumDiaSourcesSelection` (or whatever better name you can come up with, naming is hard). It could look something like:

```python
class NumDiaSourcesSelection(AnalysisTool):
    metricName = Field[str](doc="Name to use for output metric")

    def setDefaults(self):
        super().setDefaults()
        self.process.calculateActions.countingAction = <name_of_combined_action>
        self.produce.metric.units = {"countingAction": "ct"}

    def finalize(self):
        self.produce.metric.newNames = {"countingAction": self.metricName}
```

You can then use this in pipelines to setup whatever you want without needing duplicate tools, or any new tools if you want to introduce new metrics in the future. This would look something like:

```yaml
      atools.numDiaSourcesHighReliability: NumDiaSourcesSelection
      # this could be a different name if you wanted
      atools.numDiaSourcesHighReliability.metricName: numDiaSourcesHighReliability
      atools.numDiaSourcesHighReliability.process.calculateActions.countingAction.op: gt
      atools.numDiaSourcesHighReliability.process.calculateActions.countingAction.threshold: 0.9
      atools.numDiaSourcesHighReliability.process.calculateActions.countingAction.vectorKey: reliability

      atools.numDiaSourcesLowReliability: NumDiaSourcesSelection
      atools.numDiaSourcesLowReliability.metricName: numDiaSourcesLowReliability
      atools.numDiaSourcesLowReliability.process.calculateActions.countingAction.op: lt
      atools.numDiaSourcesLowReliability.process.calculateActions.countingAction.threshold: 0.1
      atools.numDiaSourcesLowReliability.process.calculateActions.countingAction.vectorKey: reliability

      atools.numDiaSourcesNanReliability: NumDiaSourcesSelection
      atools.numDiaSourcesNanReliability.metricName: numDiaSourcesNanReliability
      atools.numDiaSourcesNanReliability.process.calculateActions.countingAction.op: eq
      atools.numDiaSourcesNanReliability.process.calculateActions.countingAction.threshold: NaN
      atools.numDiaSourcesNanReliability.process.calculateActions.countingAction.vectorKey: reliability

```

This way you can have any number of metrics all defined by the configuration of one tool. There are any number of ways to do this of course, you could also have specified your config in one line in a python block, you could have forwarded more properties to make each line smaller, etc. This gives you some scope of what is possible.

As I say what you have is not wrong, and has the benifit of an importable configuration independent of a pipeline, but it does add more overhead for introducing new tools.